retriever/retriever.php

```diff
@@ -0,0 +1,1028 @@
+<?php
+  /**
+   * Name: Retriever
+   * Description: Follow the permalink of RSS/Atom feed items and replace the summary with the full content.
+   * Version: 1.0
+   * Author: Matthew Exon <http://mat.exon.name>
+   */
+
+use Friendica\Core\Addon;
+use Friendica\Core\Config;
+use Friendica\Core\PConfig;
+use Friendica\Core\Logger;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Content\Text\HTML;
+use Friendica\Content\Text\BBCode;
+use Friendica\Model\Photo;
+use Friendica\Object\Image;
+use Friendica\Util\Network;
+use Friendica\Core\L10n;
+use Friendica\Database\DBA;
+use Friendica\Model\ItemURI;
+use Friendica\Model\Item;
+use Friendica\Util\DateTimeFormat;
+
+/**
+ * @brief Installation hook for retriever plugin
+ */
+function retriever_install() {
+	Addon::registerHook('addon_settings', 'addon/retriever/retriever.php', 'retriever_addon_settings');
+	Addon::registerHook('addon_settings_post', 'addon/retriever/retriever.php', 'retriever_addon_settings_post');
+	Addon::registerHook('post_remote', 'addon/retriever/retriever.php', 'retriever_post_remote_hook');
+	Addon::registerHook('contact_photo_menu', 'addon/retriever/retriever.php', 'retriever_contact_photo_menu');
+	Addon::registerHook('cron', 'addon/retriever/retriever.php', 'retriever_cron');
+
+	if (Config::get('retriever', 'dbversion') != '0.14') {
+		$schema = file_get_contents(dirname(__file__).'/database.sql');
+		$tables = explode(';', $schema);
+		foreach ($tables as $table) {
+			if (!DBA::e($table)) {
+				Logger::warning('Unable to create database table: ' . DBA::errorMessage());
+				return;
+			}
+		}
+		Config::set('retriever', 'downloads_per_cron', '100');
+		Config::set('retriever', 'dbversion', '0.14');
+	}
+}
+
+/**
+ * @brief Uninstallation hook for retriever plugin
+ */
+function retriever_uninstall() {
+	Addon::unregisterHook('addon_settings', 'addon/retriever/retriever.php', 'retriever_addon_settings');
+	Addon::unregisterHook('addon_settings_post', 'addon/retriever/retriever.php', 'retriever_addon_settings_post');
+	Addon::unregisterHook('post_remote', 'addon/retriever/retriever.php', 'retriever_post_remote_hook');
+	Addon::unregisterHook('addon_settings', 'addon/retriever/retriever.php', 'retriever_addon_settings');
+	Addon::unregisterHook('addon_settings_post', 'addon/retriever/retriever.php', 'retriever_addon_settings_post');
+	Addon::unregisterHook('contact_photo_menu', 'addon/retriever/retriever.php', 'retriever_contact_photo_menu');
+	Addon::unregisterHook('cron', 'addon/retriever/retriever.php', 'retriever_cron');
+}
```

Comment:
Uninstalling an addon now unregisters all hooks by default, `{addon}_uninstall` is only needed for non-hook related actions on unintall.